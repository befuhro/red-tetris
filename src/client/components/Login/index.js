import React from 'react';

import PopUp from './PopUp';

const pageStyle = {
    position: 'fixed',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
};

const blockStyle = {
    padding: '20px',
    backgroundColor: '#0b1d2d',
};

const fontStyle = {
    margin: '3vh 0 1vh 0',
    color: '#61dafb',
    fontFamily: 'Permanent Marker',
    fontSize: '3vh',
    fontWeight: '300',
    textAlign: 'center',
};

const buttonStyle = {
    width: '15vh',
    marginTop: '1vh',
    fontSize: '2vh',
    fontWeight: '300',
    textAlign: 'center',
};

const form = {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center'
};

const authenticate = (e, checkAvailability) => {
    const username = document.getElementById("username").value;
    const room = document.getElementById("room").value;

    e.preventDefault();
    checkAvailability(username, room);
};

export default ({errors, checkAvailability, deleteErrors}) => {
    return (
        <div style={pageStyle}>
            <div style={blockStyle}>
                <form style={form} onSubmit={(e) => authenticate(e, checkAvailability)}>
                    <label style={fontStyle}>
                        ENTER A USERNAME
                    </label>
                    <input type="text" id="username" required/>
                    <label style={fontStyle}>
                        ENTER A ROOM
                    </label>
                    <input type="text" id="room" required/>
                    <button style={buttonStyle} type="submit">Join game</button>
                </form>
                {errors ?
                    <PopUp
                        errors={errors}
                        deleteErrors={deleteErrors}
                    /> :
                    null
                }
            </div>
        </div>
    );
}
