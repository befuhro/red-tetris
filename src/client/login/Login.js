import React from 'react';
import {connect} from 'react-redux';

import {checkAvailability, deleteErrors} from './loginActions';

const blockStyle = {
    padding: '20px',
    backgroundColor: '#0b1d2d',
};

const fontStyle = {
    margin: '3vh 0 1vh 0',
    color: '#61dafb',
    fontFamily: 'Permanent Marker',
    fontSize: '3vh',
    fontWeight: '300',
    textAlign: 'center',
};

const buttonStyle = {
    width: '15vh',
    marginTop: '1vh',
    fontSize: '2vh',
    fontWeight: '300',
    textAlign: 'center',
};

const backgroundPopUp = {
    backgroundColor: 'rgba(255, 255, 255, 0.5)',
    position: 'fixed',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
};

const popUp = {
    position: 'absolute',
    display: 'flex',
    flexDirection: 'column',
    padding: '10px',
    backgroundColor: '#0b1d2d',
};

const errorFont = {
    color: '#61dafb',
    fontFamily: 'sans-serif',
    margin: '10px 20px 10px 20px',
};

const crossContainer = {
    alignSelf: 'flex-end',
    backgroundColor: 'white',
    height: '30px',
    width: '30px',
    borderRadius: '15px'
};

const cross = {
    fontSize: '20px',
    color: 'red',
    fontFamily: 'sans-serif',
    margin: '0'
};

const form = {
    display: 'flex',
    flexDirection: 'column',
    alignItems:  'center'
};

function authenticate(e, checkAvailability) {
    const username = document.getElementById("username").value;
    const room = document.getElementById("room").value;

    e.preventDefault();
    checkAvailability(username, room);
}

function PopUp({errors, deleteErrors}) {
    return (
        <div style={backgroundPopUp}>
            <div style={popUp}>
                <button style={crossContainer}>
                    <p style={cross} onClick={deleteErrors}>x</p>
                </button>
                {errors.map((error) => <p key={error.id} style={errorFont}>{error.message}</p>)}
            </div>
        </div>
    )
}

function Login({errors, checkAvailability, deleteErrors}) {
    return (
        <div style={blockStyle}>
            <form style={form} onSubmit={(e) => authenticate(e, checkAvailability)}>
                <label style={fontStyle}>
                    ENTER A USERNAME
                </label>
                <input type="text" id="username" required/>
                <label style={fontStyle}>
                    ENTER A ROOM
                </label>
                <input type="text" id="room" required/>
                <button style={buttonStyle} type="submit">Join game</button>
            </form>
            {errors !== null && errors !== undefined ?
                <PopUp
                    errors={errors}
                    deleteErrors={deleteErrors}
                /> :
                null
            }
        </div>
    );
}

const mapStateToProps = (state) => {
    const errors = state.login.errors;
    return ({errors});
};

const mapDispatchToProps = (dispatch) => ({
    checkAvailability: (userName, room) => dispatch(checkAvailability(userName, room)),
    deleteErrors: () => dispatch(deleteErrors())
});

export default connect(mapStateToProps, mapDispatchToProps)(Login);