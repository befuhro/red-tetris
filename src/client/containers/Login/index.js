import {connect} from 'react-redux';

import Login from '../../components/Login';
import {checkAvailability, deleteErrors} from "../../actions/env";

const mapStateToProps = (state) => (
    {errors: state.login.errors}
);

const mapDispatchToProps = (dispatch) => ({
    checkAvailability: (userName, room) => dispatch(checkAvailability(userName, room)),
    deleteErrors: () => dispatch(deleteErrors()),
});

export default connect(mapStateToProps, mapDispatchToProps)(Login);
