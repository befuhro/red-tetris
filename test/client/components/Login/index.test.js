import * as React from 'react';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

enzyme.configure({ adapter: new Adapter() });

import Login from '../../../../src/client/components/Login';

describe('<Login />', () => {
   describe('no click', () => {
       let component;

       beforeAll(() => {
           component = enzyme.mount(<Login
               checkAvailability={() => {}}
               deleteErrors={() => {}}
               errors={[{id:0, message:'room is full'}]}
           />);
       });

       it('basic test', () => {
           expect(component.html()).toMatchSnapshot();
       });
   });
});
