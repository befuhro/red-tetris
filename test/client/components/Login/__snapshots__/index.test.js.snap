// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> no click basic test 1`] = `"<div style=\\"position: fixed; display: flex; flex-direction: column; justify-content: center; align-items: center; top: 0px; left: 0px; right: 0px; bottom: 0px;\\"><div style=\\"padding: 20px; background-color: rgb(11, 29, 45);\\"><form style=\\"display: flex; flex-direction: column; align-items: center;\\"><label style=\\"margin: 3vh 0px 1vh 0px; color: rgb(97, 218, 251); font-family: Permanent Marker; font-size: 3vh; font-weight: 300; text-align: center;\\">ENTER A USERNAME</label><input type=\\"text\\" id=\\"username\\" required=\\"\\"><label style=\\"margin: 3vh 0px 1vh 0px; color: rgb(97, 218, 251); font-family: Permanent Marker; font-size: 3vh; font-weight: 300; text-align: center;\\">ENTER A ROOM</label><input type=\\"text\\" id=\\"room\\" required=\\"\\"><button style=\\"width: 15vh; margin-top: 1vh; font-size: 2vh; font-weight: 300; text-align: center;\\" type=\\"submit\\">Join game</button></form><div style=\\"background-color: rgba(255, 255, 255, 0.5); position: fixed; display: flex; justify-content: center; align-items: center; top: 0px; left: 0px; right: 0px; bottom: 0px;\\"><div style=\\"position: absolute; display: flex; flex-direction: column; padding: 10px; background-color: rgb(11, 29, 45);\\"><button style=\\"align-self: flex-end; background-color: white; height: 30px; width: 30px; border-radius: 15px;\\"><p style=\\"font-size: 20px; color: red; font-family: sans-serif; margin: 0px;\\">x</p></button><p style=\\"color: rgb(97, 218, 251); font-family: sans-serif; margin: 10px 20px 10px 20px;\\">room is full</p></div></div></div><div style=\\"display: flex; justify-content: center; align-items: center; position: absolute; bottom: 0px; width: 100%; height: 14.24vh; color: rgb(97, 218, 251);\\"><p style=\\"font-family: Permanent Marker; font-size: 3vh; font-weight: 300; color: rgb(97, 218, 251); margin-right: 2vh;\\">befuhro &amp; eruaud</p><img src=\\"\\" style=\\"height: 10vh; margin-left: 2vh;\\" alt=\\"le-101-logo\\"></div></div>"`;
